### DISCUSSION TITLE
IKEA 24.4.5 firmware for TRADFRI remote (E1524) might be problematic

### DISCUSSION THREAD

**Discussion by klada** [Koenkk/zigbee2mqtt#16446] [Devices]
Does anyone have success pairing a TRADFRI remote (E1524) with the latest official firmware (24.4.5)?

The device itself should be supported, but the new firmware seems to prevent pairing with Z2M. I am always getting "interview failed" during the pairing process.

Does anybody have similar issues or is it just me?

Note: I have a couple of these devices. The ones with older firmware can pair just fine, it's the one with 24.4.5 which is causing issues.

**Comment by paulz-gh** [comment]
I have upgraded 2 of them yesterday but I only got them paired after a few tries. 
Keeping pressing/releasing a button on the remote to keep it awake while pairing is in progress seems to help.
I do have some strange issues remaining with this version. I only see action=toggle in z2m logs even when I click other buttons. But when I bind them to Ikea lamp it does seem to respond to the other buttons.

**Reply by klada** [reply]
> I have upgraded 2 of them yesterday but I only got them paired after a few tries.

Did you remove them from Z2M after upgrading to latest firmware?

**Reply by paulz-gh** [reply]
Yes I removed them before repairing them. Had to recreate the bindings after repairing.

**Reply by klada** [reply]
@paulz-gh If some buttons do not report events this sounds like a binding issue to me. Have you tried binding the LevelCtrl and/or Scenes cluster to the coordinator?

With the help of your hint regarding timing and button presses I finally got this remote working (see my other post). For me all button-presses are reported now after binding LevelCtrl and Scenes clusters.

**Reply by paulz-gh** [reply]
Yes it's bound to the coordinator for those clusters but only the toggle action arrives.
I have it bound to an IKEA light as well and that does respond to the brightness events so for the moment I'm fine but it's something that could be better. 
And the battery percentages are broken ( it shows 180% at the moment) but that's also low impact/priority.

**Comment by klada** [comment]
A short update from my side: I got it to pair and configure properly now. It was not easy. These are the steps which were necessary with the new firmware:

1. Put Z2M in pairing mode
1. Reset the device through the usual 4 presses of the inside button
2. The red light will flash for a while. Wait until the flashing is done
3. Short-press the power button once. The red light will come on again and after a few seconds Z2M will say that a new device has joined the network
4. As soon as Z2M says "Interviewing device" long-press the power button again and keep it pressed for ~10 seconds. Interviewing should now be successful
5. After the interview is complete Z2M will start configuring the device. Long-press the power button again for ~10 seconds and configuration should complete successfully

This used to be much easier with older FW versions. But after performing these steps the device is finally working properly (all button-presses are reported, battery-level is reported, remote can be bound to other ZigBee lights).

If anybody else is having issues with these devices and the new firmware, please try the steps above and report back. We might need to update the docs and add the new procedure.

**Reply by abjorck** [reply]
I tried these steps a few times but did not succeed :( 

I just gave up and paired it directly to the light for now
